tests/unautomated/test_laser_envelope.py

```diff
@@ -0,0 +1,396 @@
+        print('The simulation results agree with the theory to %e.' %rtol)
+
+    # Return a dictionary of the results
+    return( { 'A' : A, 'w' : w, 'fld' : sim.fld } )
+
+
+def init_fields( sim, w, ctau, k0, z0, zf, E0, m=1 ) :
+    """
+    Imprints the appropriate profile on the fields of the simulation.
+
+    Parameters
+    ----------
+    sim: Simulation object from fbpic
+
+    w : float
+       The initial waist of the laser (in microns)
+
+    ctau : float
+       The initial temporal waist of the laser (in microns)
+
+    k0 : flat
+       The central wavevector of the laser (in microns^-1)
+
+    z0 : float
+       The position of the centroid on the z axis
+
+    zf : float
+       The position of the focal plane
+
+    E0 : float
+       The initial E0 of the pulse
+
+    m: int, optional
+        The mode on which to imprint the profile
+        For m = 0 : gaussian profile, linearly polarized beam
+        For m = 1 : annular profile
+    """
+    # Initialize the fields
+    a0 = E0*e/(m_e*c**2*k0)
+    tau = ctau/c
+    lambda0 = 2*np.pi/k0
+    # Create the relevant laser profile
+
+    if m == 0:
+        profile = GaussianLaser( a0=a0, waist=w, tau=tau,
+                    lambda0=lambda0, z0=z0, zf=zf )
+    elif m == 1:
+        profile = LaguerreGaussLaser( 0, 1, a0=a0, waist=w, tau=tau,
+                    lambda0=lambda0, z0=z0, zf=zf )
+   
+    # Add the profiles to the simulation
+    add_laser_pulse( sim, profile, method = 'direct_envelope' )
+
+def gaussian_transverse_profile( r, w, E ) :
+    """
+    Calculte the Gaussian transverse profile.
+
+    This is used for the fit of the fields
+
+    Parameters
+    ----------
+    r: 1darray
+       Represents the positions of the grid in r
+
+    w : float
+       The initial waist of the laser (in microns)
+
+    E : float
+       The E0 of the pulse
+    """
+    return( E*np.exp( -r**2/w**2 ) )
+
+def annular_transverse_profile( r, w, E ) :
+    """
+    Calculte the annular transverse profile.
+
+    This is used both for the initialization and
+    for the fit of the fields
+
+    Parameters
+    ----------
+    r: 1darray
+       Represents the positions of the grid in r
+
+    w : float
+       The initial waist of the laser (in microns)
+
+    E : float
+       The E0 of the pulse
+    """
+    return( E*(r/w)*np.exp( -r**2/w**2 ) )
+
+def fit_fields( fld, m ) :
+    """
+    Extracts the waist and a0 of the pulse through a transverse Gaussian fit.
+
+    The laser oscillations are first averaged longitudinally.
+
+    Parameters
+    ----------
+    fld : Fields object from fbpic
+
+    m : int, optional
+       The index of the mode to be fitted
+    """
+    # Integrate the laser oscillations longitudinally
+    dz = fld.interp[0].dz
+    laser_profile = np.sqrt(dz*(abs( fld.envelope_interp[m].A )**2).sum(axis=0)) 
+    # Renormalize so that this gives the peak of the Gaussian
+    laser_profile *= 2.**(-3./4)/( np.pi**(1./4) * ctau**(1./2) )
```

Comment:
Actually, the aim of the 3 above lines was originally to evaluate the amplitude of the laser envelope, in the case where the available data is an oscillating electric field.

However, here, the available data is the laser envelope directly. So we could replace these 3 lines by simply
```
laser_envelope = abs( fld.envelope_interp[m].A )
```

Reply:
I'm pretty sure this method wouldn't work since we actually need to make the z dependency disappear in the array.
I just tried evaluating the envelope at the estimated center of the beam in the z direction then fit the curve ( something like laser_profile = abs(fld.envelope_interp[m].A[izref]) with izref well chosen) which seems to work pretty well